core/src/main/scala/org/scalajs/macrotaskexecutor/MacrotaskExecutor.scala

```diff
@@ -67,17 +75,15 @@ object MacrotaskExecutor extends ExecutionContextExecutor {
         if (currentlyRunningATask) {
           js.Dynamic.global.setTimeout(() => runIfPresent(handle), 0)
         } else {
-          tasksByHandle.get(handle) match {
-            case Some(task) =>
-              currentlyRunningATask = true
-              try {
-                task()
-              } finally {
-                tasksByHandle -= handle
-                currentlyRunningATask = false
-              }
-
-            case None =>
+          val task = tasksByHandle(handle)
```

Comment:
With `tasksByHandle(handle)` now returning a `Runnable`, there will be an implicit cast to `Runnable` here that will CCE when it's `undefined`.

With a custom `TaskMap` (which really makes sense, btw), you'll get a better mileage by having the `apply` return a `js.UndefOr[Runnable]`. Then you could actually just call `foreach` on it, and you won't need the cast further down. Seems like a win overall.

I wonder how we could better test that each code path actually works. The code paths that are polyfilling really old stuff are under-tested, it seems.

Reply:
> I wonder how we could better test that each code path actually works. The code paths that are polyfilling really old stuff are under-tested, it seems.

Actually, I think the reason we are not hitting the CCE in the `undefined` case is because it's impossible for us. I don't think we need to consider this case at all.

It was cargo-culted from the upstream, but the upstream also implements `clearImmediate`.
https://github.com/YuzuJS/setImmediate/blob/f1ccbfdf09cb93aadf77c4aa749ea554503b9234/setImmediate.js#L31-L33